src/reportengine/resourcebuilder.py

```diff
@@ -142,11 +142,11 @@ def __init__(self, graph, rootns, environment=None):
         self.environment = environment
         self._node_flags = defaultdict(lambda: set())
 
-    def resolve_callargs(self, callspec):
+    def resolve_callargs(self, callspec, perform_final=True):
```

Comment:
`perform_final` might want to be a slightly more global variable in the class to avoid repeating

Reply:
Would be best to make it an instance-level property.

Reply:
I do think making this an instace property is better than passing it around everywhere.